### ISSUE TITLE
Admin - Data Model - Generic Date/time semantic types

### ISSUE THREAD

**Issue by chucksense** [metabase/metabase#15963]
**Is your feature request related to a problem? Please describe.**
The date and time semantic types have a bunch of things like "Created", "Updated", "Joined", "Cancellation", etc.  While these are nice, there are many times you have a date that doesn't mean any of those and you have to pick one arbitrarily to still get the Metabase date filter / aggregation features.

**Describe the solution you'd like**
Generic date and timestamp semantic types would be nice to have so you aren't prescribing incorrect meaning to something.

**Describe alternatives you've considered**
Continue to arbitrarily pick one that implies incorrect meaning.

**How important is this feature to you?**
Nice to have. It works as it is today, just annoying.

**Comment by dpsutton** [comment]
@chucksense 
I'm updated the semantic type of the "created_at" field of the "orders" table in the sample dataset to no semantic type. I'm still able to use date filters
<img width="1064" alt="image" src="https://user-images.githubusercontent.com/6377293/117478213-7ca7fe80-af24-11eb-86f3-724e611f3d48.png">
and aggregations on that field
<img width="1543" alt="image" src="https://user-images.githubusercontent.com/6377293/117478264-8cbfde00-af24-11eb-9342-058f160b002f.png">

Here is where i marked that field as having no special type:
<img width="620" alt="image" src="https://user-images.githubusercontent.com/6377293/117478311-977a7300-af24-11eb-9869-b584aaf485d0.png">

Can you give me some information about the field that is behaving this way for you?

**Comment by flamber** [comment]
This is similar to the discussion in #9202.

Given the major changes in 0.39 with the shift from `special_type`, which had multiple meanings, to `semantic_type` that is only lightly used in X-rays, then I think it's time to revise if we should remove some/most of these specific types in favor of generic types (or possibly none).
In many of the real-world databases I used to work with, the specific types made very little sense.

**Comment by dpsutton** [comment]
I don't follow what you mean by remove the specific types in favor of generic types?

**Comment by flamber** [comment]
Replace "Creation date" etc with a generic "Date", or at least have that as an option. But it depends on how that could affect X-ray

**Comment by dpsutton** [comment]
That's the point of "no semantic type". The thing that dictates that this is a date is the effective_type. The semantic type exists if you want to add some semantics to what kind of date, but the fact that it is a date comes from the `effective_type`. That's why i'm interested to see what the field information and perhaps even database are in this example.

**Comment by flamber** [comment]
That's not how things have worked for many years and @chucksense like many others has been used to having to define Field Type to get the formatting to apply correctly. And since that is not needed anymore since 0.39, then we should probably move the Field Type to only be shown on the field details page.

**Comment by chucksense** [comment]
You are correct, the filters are indeed working as expected under 0.39 as long as the underlying column is a date type.  But it doesn't seem to work if the underlying column is a string without explicitly applying a semantic type.  Is this what "Cast to a specific data type" would do instead?

**Comment by dpsutton** [comment]
Yes. There are a few ways to cast specific types. Go to admin > Datamodel , find the db, and then the table and field you want to cast. Click on the cog and then find the section to set a cast:

<img width="380" alt="image" src="https://user-images.githubusercontent.com/6377293/117689317-38fd0100-b17f-11eb-8893-9174572d9407.png">.

**Comment by thebiglabasky** [comment]
May I assume this is something we can close? There might still be places where we need semantic types, but the effective type allowing to operate with dates seems to make this issue no longer relevant?

**Comment by paoliniluis** [comment]
@thebiglabasky we will use the semantic types at some point in the future (I guess?) but we should really try to reduce complexity

**Comment by thebiglabasky** [comment]
Agreed about reducing complexity. The way things are laid out doesn't seem to give people enough clarity as to our distinction between semantic types and data types (+ coercions). But having a generic semantic type of the root data type doesn't seem like a good idea since it actually adds to the confusion rather than the other way around.<br>So I would like to address the root issue so that people can clearly understand that as long as a column is a Date (either coming straight from the DB, or cast into one), then we'll apply Date treatments to the field throughout, and not require any additional semantics if they don't want to.

**Comment by thebiglabasky** [comment]
After looking into our path forward, I'll close this issue: we're going to work out the UX of editing metadata so that it's crystal clear that the underlying data type can be *refined* using semantic types, but that semantic types aren't controlling the underlying data type. They're decorations essentially. So anything that's already a basic data type (like Date) won't get a separate semantic type on top, since that semantic type would be 1:1 the data type, hence not enriching anything.
